app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx

```diff
@@ -0,0 +1,191 @@
+import { useContext, useMemo, useState } from "react";
+
+import { ArboContext } from "@/contexts/arbo-context/arbo-context";
+import {
+  mixColours,
+  typedGroupBy,
+  typedObjectEntries,
+  typedObjectFromEntries,
+  typedObjectKeys,
+} from "@/lib/utils";
+import { arbovirusesSF, convertArboSFtoArbo, median } from "./recharts";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { WHORegion } from "@/lib/who-regions";
+import { Button } from "@/components/ui/button";
+
+enum AgeGroup {
+  "Adults (18-64 years)" = "Adults (18-64 years)",
+  "Children and Youth (0-17 years)" = "Children and Youth (0-17 years)",
+  "Seniors (65+ years)" = "Seniors (65+ years)",
+  "Multiple groups" = "Multiple groups",
+}
+
+export const MedianSeroprevalenceByWhoRegionAndAgeGroupTable = () => {
+  const state = useContext(ArboContext);
+  const pathogenOrder: arbovirusesSF[] = [
+    "ZIKV",
+    "DENV",
+    "CHIKV",
+    "YF",
+    "WNV",
+    "MAYV",
+  ];
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const datasetGroupedByArbovirus = useMemo(
+    () =>
+      typedGroupBy(
+        state.filteredData,
+        (dataPoint) => dataPoint.pathogen as arbovirusesSF
+      ),
+    [state.filteredData]
+  );
+
+  const datasetGroupedByArbovirusAndWhoRegion = useMemo(
+    () =>
+      typedObjectFromEntries(
+        typedObjectEntries(datasetGroupedByArbovirus).map(
+          ([arbovirus, dataPoints]) => [
+            arbovirus,
+            typedGroupBy(
+              dataPoints,
+              (dataPoint) => dataPoint.whoRegion as WHORegion
+            ),
+          ]
+        )
+      ),
+    [datasetGroupedByArbovirus]
+  );
+
+  const tableDatasets = useMemo(
+    () =>
+      typedObjectFromEntries(
+        typedObjectEntries(datasetGroupedByArbovirusAndWhoRegion).map(
+          ([arbovirus, dataGroupedByWHORegion]) => [
+            arbovirus,
+            typedObjectFromEntries(
+              typedObjectEntries(dataGroupedByWHORegion).map(
+                ([whoRegion, dataPoints]) => [
+                  whoRegion,
+                  typedGroupBy(
+                    dataPoints,
+                    (dataPoint) => dataPoint.ageGroup as AgeGroup
+                  ),
+                ]
+              )
+            ),
+          ]
+        )
+      ),
+    [datasetGroupedByArbovirusAndWhoRegion]
+  );
+
+  const currentlySelectedArbovirus = useMemo(
+    () => pathogenOrder[currentIndex],
+    [pathogenOrder, currentIndex]
+  );
+
+  const datasetToDisplay = useMemo(
+    () => tableDatasets[currentlySelectedArbovirus],
+    [currentlySelectedArbovirus, tableDatasets]
+  );
+
+  return (
+    <div className="p-2">
+      <div className="flex justify-between mb-2 ignore-for-visualization-download">
```

Comment:
Don't want the header and the buttons to get included in the downloaded png. I'll swap over the buttons to use this same method at some point, not this PR though.